src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -704,10 +704,55 @@ namespace winrt::TerminalApp::implementation
         if (_tabs.Size() == 0 && !(_shouldStartInboundListener || _isEmbeddingInboundListener))
         {
             _LastTabClosedHandlers(*this, nullptr);
+            co_return;
         }
         else
         {
-            _InitializedHandlers(*this, nullptr);
+            // GH#11561: When we start up, our window is initially just a frame
+            // with a transparent content area. We're gonna do all this startup
+            // init on the UI thread, so the UI won't actually paint till it's
+            // all done. This results in a few frames where the frame is
+            // visible, before the page paints for the first time, before any
+            // tabs appears, etc.
+            //
+            // To mitigate this, we're gonna wait for the UI thread to finish
+            // everything it's gotta do for the initial init, and _then_ fire
+            // our Initialized event. By waiting for everything else to finish
+            // (CoreDispatcherPriority::Low), we let all the tabs and panes
+            // actually get created. In the window layer, we're gonna cloak the
+            // window till this event is fired, so we don't actually see this
+            // frame until we're actually all ready to go.
+            //
+            // This will result in the window seemingly not loading as fast, but
+            // it will actually take exactly the same amount of time before it's
+            // usable.
+            //
+            // We also experimented with drawing a solid BG color before the
+            // initialization is finished. However, there are still a few frames
+            // after the frame is displayed before the XAML content first draws,
+            // so that didn't actually resolve any issues.
+
+            auto weak{ get_weak() };
+
+            // Switch to the BG thread -
+            co_await winrt::resume_background();
+
+            if (auto self{ weak.get() })
+            {
+                // Then enqueue the rest of this function for after the UI thread settles.
+                co_await wil::resume_foreground(self->Dispatcher(), CoreDispatcherPriority::Low);
+            }
+            else
+            {
+                // We don't exist anymore? Well, we probably don't need to fire
+                // off an Initialized event then...
+                co_return;
+            }
+
+            if (auto self{ weak.get() })
+            {
+                self->_InitializedHandlers(*self, nullptr);
+            }
```

Comment:
I think something like this is equivalent but much shorter:
```suggestion
            Dispatcher().RunAsync(CoreDispatcherPriority::Low, [weak = get_weak()]() {
                if (auto self{ weak.get() })
                {
                    self->_InitializedHandlers(*self, nullptr);
                }
            });
```

Reply:
Yoo I agree let's do it. Sometimes coroutines are the answer! Sometimes they are not.

Reply:
(This also compiles to fewer coroutine funclets!)